test/unit/ngsiv2/ngsiService/subscriptions-test.js

```diff
@@ -0,0 +1,351 @@
+
+                        if (JSON.stringify(expected) === JSON.stringify(body))
+                        {
+                            return true;
+                        }
+                        else {
+                            return false;
+                        }
+                    }
+                    else {
+                        return false;
+                    }
+                })
+                .reply(201, null, {'Location': '/v2/subscriptions/51c0ac9ed714fb3b37d7d5a8'});
+
+            iotAgentLib.clearAll(function() {
+                request(optionsProvision, function(error, result, body) {
+                    done();
+                });
+            });
+        });
+    });
+
+    afterEach(function(done) {
+        nock.cleanAll();
+        iotAgentLib.setNotificationHandler();
+        iotAgentLib.clearAll(function() {
+            iotAgentLib.deactivate(done);
+        });
+    });
+
+    describe('When a client invokes the subscribe() function for device', function() {
+        it('should send the appropriate request to the Context Broker', function(done) {
+            iotAgentLib.getDevice('MicroLight1', 'smartGondor', '/gardens', function(error, device) {
+
+                iotAgentLib.subscribe(device, ['attr_name'], null, function(error) {
+                    should.not.exist(error);
+
+                    contextBrokerMock.done();
+
+                    done();
+                });
+            });
+        });
+        it('should store the subscription ID in the Device Registry', function(done) {
+            iotAgentLib.getDevice('MicroLight1', 'smartGondor', '/gardens', function(error, device) {
+                iotAgentLib.subscribe(device, ['attr_name'], null, function(error) {
+                    iotAgentLib.getDevice('MicroLight1', 'smartGondor', '/gardens', function(error, device) {
+                        should.not.exist(error);
+                        should.exist(device);
+                        should.exist(device.subscriptions);
+                        device.subscriptions.length.should.equal(1);
+                        device.subscriptions[0].id.should.equal('51c0ac9ed714fb3b37d7d5a8');
+                        device.subscriptions[0].triggers[0].should.equal('attr_name');
+                        done();
+                    });
+                });
+            });
+        });
+    });
+    describe('When a client invokes the unsubscribe() function for an entity', function() {
+        beforeEach(function(done) {
+            contextBrokerMock = nock('http://192.168.1.1:1026')
+                .matchHeader('fiware-service', 'smartGondor')
+                .matchHeader('fiware-servicepath', '/gardens')
+                .delete('/v2/subscriptions/51c0ac9ed714fb3b37d7d5a8')
+                .reply(204);
+
+            done();
+        });
+        it('should change the expiration date of the subscription to 0s', function(done) {
```

Comment:
I'm a bit lost in the callback chain :)... how this step ensures expiration date of the subscrition is set to 0?

Reply:
In fact, one doubt: "expiration date" here referst to the expiration date that Orion manages? Or some internal data in the IOTA database?

Reply:
It does not check anything regarding this "expiration date". I just copied the test from https://github.com/telefonicaid/iotagent-node-lib/blob/master/test/unit/ngsiService/subscriptions-test.js#L176 and I guess that it is probably not correct. The only thing that this test does is to check if the mock declare in L165 is called (unsusbcribe in the CB).

Reply:
In this case I think it would be better to change the name of this and the original test in order to have some more meaninful. For instance:

```
should change unsubscription takes place
```
or some more creative wording in the same line

Reply:
Fixed in https://github.com/dcalvoalonso/iotagent-node-lib/pull/4/commits/3c52ae9bf9547efd7b195db5219f818882dc1030